libraries/rush-sdk/src/proxy.ts

```diff
@@ -0,0 +1,288 @@
+   *
+   * @remarks
+   * If a long-running operation is required, then `progressBarPercent` will
+   * start at 0.0 and count upwards and finish at 100.0 if the operation completes
+   * successfully.  If the long-running operation has not yet started, or
+   * is not required, then the value will be `undefined`.
+   */
+  progressBarPercent: number | undefined;
+}
+
+/**
+ * Type of {@link ILoadSdkAsyncOptions.onNotifyEvent}
+ * @public
+ */
+export type SdkNotifyEventCallback = (sdkEvent: ISdkCallbackEvent) => void;
+
+/**
+ * Options for {@link rushSdkProxy.loadAsync}
+ * @public
+ */
+export interface ILoadSdkAsyncOptions {
+  /**
+   * The folder to start from when searching for the Rush workspace configuration.
+   * If this folder does not contain a `rush.json` file, then each parent folder
+   * will be searched.  If `rush.json` is not found, then the SDK fails to load.
+   */
+  rushJsonSearchFolder?: string;
+
+  /**
+   * A cancellation token that the caller can use to prematurely abort the operation.
+   */
+  abortSignal?: AbortSignal;
+
+  /**
+   * Allows the caller to monitor the progress of the operation.
+   */
+  onNotifyEvent?: SdkNotifyEventCallback;
+}
+
+/**
+ * Exposes operations that control how the `@microsoft/rush-lib` engine is
+ * located and loaded.
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace rushSdkProxy {
+  function _checkForCancel(
+    abortSignal: AbortSignal,
+    onNotifyEvent: SdkNotifyEventCallback | undefined,
+    progressBarPercent: number | undefined
+  ): boolean {
+    if (!abortSignal.aborted) {
+      return true;
+    }
+
+    if (onNotifyEvent) {
+      onNotifyEvent({
+        state: 'aborted',
+        logMessage: {
+          messageType: 'info',
+          message: `The operation was cancelled`
+        },
+        progressBarPercent
+      });
+    }
+
+    return false;
+  }
+
+  /**
+   * Manually load the Rush engine based on rush.json found for `rushJsonSearchFolder`.
+   * This API supports an optional cancellation token and callback for status updates
+   * during the operation.
+   */
+  export async function loadAsync(options?: ILoadSdkAsyncOptions): Promise<void> {
+    // SCENARIO 5: The rush-lib engine is loaded manually using rushSdkProxy.loadAsync().
+
+    if (!options) {
+      options = {};
+    }
+
+    const onNotifyEvent: SdkNotifyEventCallback | undefined = options.onNotifyEvent;
+    let progressBarPercent: number | undefined = undefined;
+
+    const abortSignal: AbortSignal = options.abortSignal ?? { aborted: false };
+
+    try {
+      const rushJsonSearchFolder: string = options.rushJsonSearchFolder ?? process.cwd();
+      const rushJsonPath: string | undefined = tryFindRushJsonLocation(rushJsonSearchFolder);
+      if (!rushJsonPath) {
+        throw new Error(
+          'Unable to find rush.json in the current folder or its parent folders.\n' +
```

Comment:
Since `options.rushJsonSearchFolder` is not required to be `process.cwd()`, this error should log the value of `rushJsonSearchFolder`. Should also update to make no claim about the working directory, since that's why the parameter exists.

Reply:
Good catch, fixed